### ISSUE TITLE
Prefabs sometimes get corrupted and lose their root actor, possibly related to engine crashes

### ISSUE THREAD

**Issue by acidiclight** [FlaxEngine/FlaxEngine#1432]
**Issue description:**

In my game, I created a prefab with a root Container Control and two child Image controls. I then had a few engine crashes, and the prefab seemed to lose its root actor.

I'd expect to open the prefab in the prefab editor and see the three actors, but I instead get an empty hierarchy that doesn't let me add new actors to it. Only way to work around it is to delete the prefab, recreate it from scratch, and re-assign any references to the old prefab.

Furthermore, I get a NullReferenceException if I try to spawn the broken prefab at runtime, i.e.

```C#
// Actual code from the attached project
PrefabManager.SpawnPrefab(backdropPrefab, this.canvasScalerActor).MustGetComponent(out backdrop);
```

Recent log files from the last few hours of work:

[Log_2023_09_18_19_42_07.txt](https://github.com/FlaxEngine/FlaxEngine/files/12654827/Log_2023_09_18_19_42_07.txt)
[Log_2023_09_18_19_33_35.txt](https://github.com/FlaxEngine/FlaxEngine/files/12654828/Log_2023_09_18_19_33_35.txt)
[Log_2023_09_18_19_16_28.txt](https://github.com/FlaxEngine/FlaxEngine/files/12654829/Log_2023_09_18_19_16_28.txt)
[Log_2023_09_18_17_39_36.txt](https://github.com/FlaxEngine/FlaxEngine/files/12654830/Log_2023_09_18_17_39_36.txt)
[Log_2023_09_19_21_44_03.txt](https://github.com/FlaxEngine/FlaxEngine/files/12654831/Log_2023_09_19_21_44_03.txt)
[Log_2023_09_18_20_20_35.txt](https://github.com/FlaxEngine/FlaxEngine/files/12654832/Log_2023_09_18_20_20_35.txt)

One of the corrupt prefabs I noticed:

`/Content/UI/Backdrop/Backdrop.prefab` (won't let me upload it directly)

**Steps to reproduce:**

Haven't figured out a reliabler way to reproduce this, seems to be random. I've attached the game with the issue though, with any sensitive code removed.

**Minimal reproduction project:**

[SociallyDistant.zip](https://github.com/FlaxEngine/FlaxEngine/files/12654810/SociallyDistant.zip)

**Flax version:**

Compiled from source this morning (Sep. 18th 2023)

**Comment by lifeformed** [comment]
I get this too.  I managed to reproduce it on an empty project:
1. Create an empty actor in the scene
2. Give it an empty actor child
3. Right click on the parent -> Create Prefab
4. Open the prefab in the editor
5. Right click on the child -> Create Prefab
6. At this point, you should get these warnings in the output:
```[ 00:00:08.729 ]: [Warning] Unable to find scripting object with ID=814f0f024252201dc168cb804da02907. Required type FlaxEngine.Actor.
[ 00:00:08.729 ]: [Warning] Missing parent actor 814f0f024252201dc168cb804da02907 for 'second (FlaxEngine.EmptyActor; 96dbcbeb419b67a23db6a194c4686001)'```
7. Open up the new child prefab in the editor
8. Change a value on it, like the position.Y, then save
10. In the first prefab, change that same value to a new value, then save
11. In the second prefab, change that same value to something else
12. Now you will be unable to save.  You get an error:

```[ 00:05:58.976 ]: [Warning] Missing object e8e9f1e440250a4ea28ae78bdcce4283 data in prefab FlaxEngine.Prefab, 2f9abd1a4cf4802387e1c5a237e41a2d, {myPath}/Content/EmptyActor.prefab.
[ 00:05:58.976 ]: [Warning] Failed to deserialize scene object from data: {
	"ID": "8067349d410784fc801681ac4215652c",
	"PrefabID": "2f9abd1a4cf4802387e1c5a237e41a2d",
	"PrefabObjectID": "e8e9f1e440250a4ea28ae78bdcce4283",
	"Transform": {
		"Translation": {
			"X": 0.0,
			"Y": 8.0,
			"Z": 0.0
		}
	}
}
[ 00:05:58.976 ]: [Warning] Missing prefab root object.
[ 00:05:58.977 ]: [Warning] Failed to create default prefab instance for the prefab asset.
[ 00:05:59.127 ]: [Warning] Exception has been thrown during Window.OnMouseUp. Failed to apply the prefab. See log to learn more.
Stack strace:
   at FlaxEditor.Modules.PrefabsModule.ApplyAll(Actor instance) in {myPath}\flax\Source\Editor\Modules\PrefabsModule.cs:line 227
   at FlaxEditor.Windows.Assets.PrefabWindow.Save() in {myPath}\flax\Source\Editor\Windows\Assets\PrefabWindow.cs:line 351
   at FlaxEditor.GUI.ToolStripButton.OnMouseUp(Float2 location, MouseButton button) in {myPath}\flax\Source\Editor\GUI\ToolStripButton.cs:line 186
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(Float2 location, MouseButton button) in {myPath}\flax\Source\Engine\UI\GUI\ContainerControl.cs:line 979
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(Float2 location, MouseButton button) in {myPath}\flax\Source\Engine\UI\GUI\ContainerControl.cs:line 979
   at FlaxEditor.Windows.Assets.PrefabWindow.OnMouseUp(Float2 location, MouseButton button) in {myPath}\flax\Source\Editor\Windows\Assets\PrefabWindow.cs:line 269
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(Float2 location, MouseButton button) in {myPath}\flax\Source\Engine\UI\GUI\ContainerControl.cs:line 979
   at FlaxEditor.GUI.Docking.DockPanelProxy.OnMouseUp(Float2 location, MouseButton button) in {myPath}\flax\Source\Editor\GUI\Docking\DockPanelProxy.cs:line 405
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(Float2 location, MouseButton button) in {myPath}\flax\Source\Engine\UI\GUI\ContainerControl.cs:line 979
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(Float2 location, MouseButton button) in {myPath}\flax\Source\Engine\UI\GUI\ContainerControl.cs:line 979
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(Float2 location, MouseButton button) in {myPath}\flax\Source\Engine\UI\GUI\ContainerControl.cs:line 979
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(Float2 location, MouseButton button) in {myPath}\flax\Source\Engine\UI\GUI\ContainerControl.cs:line 979
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(Float2 location, MouseButton button) in {myPath}\flax\Source\Engine\UI\GUI\ContainerControl.cs:line 979
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(Float2 location, MouseButton button) in {myPath}\flax\Source\Engine\UI\GUI\ContainerControl.cs:line 979
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(Float2 location, MouseButton button) in {myPath}\flax\Source\Engine\UI\GUI\ContainerControl.cs:line 979
   at FlaxEngine.GUI.ContainerControl.OnMouseUp(Float2 location, MouseButton button) in {myPath}\flax\Source\Engine\UI\GUI\ContainerControl.cs:line 979
   at FlaxEngine.Window.Internal_OnMouseUp(Float2& mousePos, MouseButton button) in {myPath}\flax\Source\Engine\Platform\Window.cs:line 242
   at FlaxEngine.Interop.NativeInterop.Invoker.InvokerNoRet2`3.MarshalAndInvoke(Object delegateContext, ManagedHandle instancePtr, IntPtr paramPtr) in {myPath}\flax\Source\Engine\Engine\NativeInterop.Invoker.cs:line 375
   at FlaxEngine.Interop.NativeInterop.InvokeMethod(ManagedHandle instanceHandle, ManagedHandle methodHandle, IntPtr paramPtr, IntPtr exceptionPtr) in {myPath}\flax\Source\Engine\Engine\NativeInterop.Unmanaged.cs:line 691
[ 00:05:59.127 ]: [Error] Exception has been thrown during Window.OnMouseUp.
Failed to apply the prefab. See log to learn more.
```

**Comment by Menotdan** [comment]
I am able to reproduce the warning, but I can't reproduce the crash. Tested on 1.6 and master.

**Comment by acidiclight** [comment]
It's possible that my crash may be unrelated to the bug itself. I only
noticed the corrupt prefab after the crash though, and I crashed when
saving the prefab. Perhaps the crash just happened coincidentally at the
same time.

That being said it's nice knowing others can reproduce the actual big. If
someone can provide a more minimal project then I will update my original
post and remove my actual game code from it.

**Comment by mafiesto4** [comment]
Fixed bug in 9870d162e4c5e86706dbc3e91ffe5aa241079e39 - now it works smoothly when creating prefab from object that is already part of the prefab.
